packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -133,12 +116,8 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
       ' or pass clientId when initializing GoogleSignIn',
     );
 
-    assert(
-      params.serverClientId == null,
-      'serverClientId is not supported on Web.',
-    );
-
-    _initCalled = Completer<void>();
```

Comment:
This may work by putting the completer in a local variable:
```dart
final initCalled = _initCalled = Completer<void>();
```

and later:
```dart
initCalled.complete(); // Use the local completer, not `_initCalled`
```